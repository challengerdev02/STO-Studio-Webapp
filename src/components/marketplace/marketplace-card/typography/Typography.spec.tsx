import { cleanup, queryByText, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { CustomStyledText } from './index';

describe('Container', function () {
  afterEach(cleanup);
  let props = {};

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(
      <CustomStyledText {...props} {...extraProps}>
        ABC
      </CustomStyledText>
    );
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should have CustomStyledText "This is a test CustomStyledText"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(queryByText(document.body, 'ABC')).toBeInTheDocument();
  });

  it('should have CustomStyledText "muted= true"', () => {
    // @ts-ignore
    const { container } = setup({ muted: true });
    expect(queryByText(document.body, 'ABC')).toBeInTheDocument();
  });
});
