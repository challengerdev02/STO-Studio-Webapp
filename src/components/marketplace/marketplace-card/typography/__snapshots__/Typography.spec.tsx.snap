// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match with snapshot 1`] = `
<div>
  <span
    class="ant-typography sc-bdvvtL cEDpem"
  >
    ABC
  </span>
</div>
`;
